### ISSUE TITLE
useless-super-delegation is emitted with different arguments

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#1043]
When a method is extended like this:

```python
    def __init__(self, none_ok=False):
        super().__init__(
            none_ok,
            valid_values=[23, 42])
```

the `super()` call actually serves a purpose, yet I get `useless-super-delegation` with the current pylint/astroid master

**Comment by PCManticore** [comment]
Yes, there seems to be a bug.

But why don't you let valid_values be passed as well into the signature of this ``__init__``? This kind of change introduces a discrepancy between how two objects behave in relationship with one another, one accepting parameters which the other one doesn't, even though one is a sub type of the other one. I think the implementation should accept at least the same parameters as the superclasses, otherwise inheritance doesn't make that much sense.

**Comment by The-Compiler** [comment]
So those are types of config values.

I have a `BaseType` and various types inheriting from it. Some of them (like `String`) take `valid_values` as there can be various settings with that type but different contraints, while for others (like `Position`) it's fixed (e.g. `top`, `bottom`, `left`, `right`).

I'm happy to other suggestions for cleaner implementations - see [configtypes.py](https://github.com/The-Compiler/qutebrowser/blob/master/qutebrowser/config/configtypes.py) and [configdata.py](https://github.com/The-Compiler/qutebrowser/blob/master/qutebrowser/config/configtypes.py) for some context.
